changelog_unreleased/javascript/12350.md

```diff
@@ -0,0 +1,2 @@
+Updated the State of JS Survey Results in order to comply with the 2021 Survey.
+I have removed unused dependecies with depcheck and where applicable, have infered parameter types from usage for type checking. (#12350 by @Harry-Hopkinson)
```

Comment:
"JavaScript" isn't the right changelog subdirectory. Turns out we don't yet have a changelog category for this kind of changes. Please add a `misc` directory (with `.gitkeep`) in `changelog_unreleased` and then add it to `scripts/draft-blog-post.mjs` and `scripts/utils/changelog-categories.mjs`.

Reply:
Do we need to have CHANGELOG at all given that the PR is only about docs? I thought CHANGELOG was related to `prettier` library only.

Reply:
I thought it might be useful for encouraging non-code contributions like this. Now if someone contributes, for example, to the docs, they won't get any credits at all when the next release happens.